packages/blocks/shuffle/src/shuffle.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useRef, useState } from "react";
+import {
+  BlockComponent,
+  useGraphBlockService,
+} from "@blockprotocol/graph/react";
+import Button from "@mui/material/Button";
+import Box from "@mui/material/Box";
+import produce from "immer";
+import { v4 as uuid } from "uuid";
+import isEqual from "lodash.isequal";
+import ShuffleIcon from "@mui/icons-material/Shuffle";
+import AddIcon from "@mui/icons-material/Add";
+import { ItemList } from "./components/item-list";
+
+export type Item = {
+  id: string;
+  value: string;
+};
+
+export type Items = Item[];
+
+type BlockEntityProperties = {
+  items?: string[];
+};
+
+export const initialItems = ["Thing 1", "Thing 2"];
+
+const createItems = (items: string[]) =>
+  items.map((item) => ({ id: uuid(), value: item }));
+
+const getItemValues = (items: Items) => items.map((item) => item.value);
+
+export const Shuffle: BlockComponent<BlockEntityProperties> = ({
+  graph: {
+    blockEntity: {
+      entityId,
+      properties: { items },
+    },
+  },
+}) => {
+  const blockRootRef = useRef<HTMLDivElement>(null);
+  const { graphService } = useGraphBlockService(blockRootRef);
+
+  const [draftItems, setDraftItems] = useState(() =>
+    createItems(items?.length ? items : initialItems),
+  );
+
+  const [prevItems, setPrevItems] = useState(items);
+
+  if (items && !isEqual(items, prevItems)) {
+    setPrevItems(items);
+    if (!isEqual(items, getItemValues(draftItems))) {
+      setDraftItems(createItems(items));
+    }
+  }
```

Comment:
I think this is fine for now, but it's worth bearing in mind that this will potentially overwrite any local changes the user has recently made, e..g if a user adds Thing 1, and the update goes to the db, and then the user adds Thing 2 and a millisecond later the update from the db with Thing 1 arrives, it will be different to the `draftItems` and they will be overwritten, removing Thing 2.

The solution to this could be some kind of local action stack (instead of just just `draftItems`), and then actions could be replayed on top of any update coming from the db. There would be more to it than that, e.g. tracking which actions had been sent to the db but hadn't been reflected in the `items` prop yet, but it's something that should be handled in the collab service rather than in individual blocks. We probably want to have updates to entity properties expressed as finer-grained actions for this reason – it's not something to fix in this PR.